optimizely/optimizely.py

```diff
@@ -160,19 +160,21 @@ def _validate_user_inputs(self, attributes=None, event_tags=None):
 
         return True
 
-    def _send_impression_event(self, project_config, experiment, variation, user_id, attributes):
+    def _send_impression_event(self, project_config, experiment, variation, flag_key, flag_type, user_id, attributes):
         """ Helper method to send impression event.
 
     Args:
       project_config: Instance of ProjectConfig.
       experiment: Experiment for which impression event is being sent.
       variation: Variation picked for user for the given experiment.
+      flag_key: key for a feature flag or experiment
+      flag_type: type for the source
```

Comment:
Why are we not doing a metadata object vs passing in so many things around?

Reply:
we only pass flag_key and flag_type around (2 things). metadata contains variation_key, flag_key,  and flag_type. 
here we pass variation and then we get id and key out of that. I think it's better to continue passing the things the way they are now